File: src/main/java/com/github/bfalmeida/photosync/model/SyncStatistics.java
package com.github.bfalmeida.photosync.model;

import java.util.concurrent.atomic.AtomicInteger;

public class SyncStatistics {
    private final AtomicInteger filesFound = new AtomicInteger(0);
    private final AtomicInteger copied = new AtomicInteger(0);
    private final AtomicInteger skipped = new AtomicInteger(0);
    private final AtomicInteger errors = new AtomicInteger(0);

    public void incrementFound() { this.filesFound.incrementAndGet(); }
    public void incrementCopied() { this.copied.incrementAndGet(); }
    public void incrementSkipped() { this.skipped.incrementAndGet(); }
    public void incrementErrors() { this.errors.incrementAndGet(); }

    public int getFilesFound() { return filesFound.get(); }
    public int getCopied() { return copied.get(); }
    public int getSkipped() { return skipped.get(); }
    public int getErrors() { return errors.get(); }

    @Override
    public String toString() {
        return String.format("Found %d files: %d copied, %d skipped, %d errors", 
            filesFound.get(), copied.get(), skipped.get(), errors.get());
    }
}

File: src/main/java/com/github/bfalmeida/photosync/model/SyncStatistics.kt
package com.github.bfalmeida.photosync.model

import java.util.concurrent.atomic.AtomicInteger

class SyncStatistics {
    private val filesFound = AtomicInteger(0)
    private val copied = AtomicInteger(0)
    private val skipped = AtomicInteger(0)
    private val errors = AtomicInteger(0)

    fun incrementFound() { filesFound.incrementAndGet() }
    fun incrementCopied() { copied.incrementAndGet() }
    fun incrementSkipped() { skipped.incrementAndGet() }
    fun incrementErrors() { errors.incrementAndGet() }

    val filesFoundCount: Int get() = filesFound.get()
    val copiedCount: Int get() = copied.get()
    val skippedCount: Int get() = skipped.get()
    val errorsCount: Int get() = errors.get()

    override fun toString(): String {
        return "Found ${filesFound.get()} files: ${copied.get()} copied, ${skipped.get()} skipped, ${errors.get()} errors"
    }
}
